lib/pure/lexbase.nim

```diff
@@ -57,17 +48,21 @@ proc fillBuffer(L: var BaseLexer) =
     oldBufLen: int
   # we know here that pos == L.sentinel, but not if this proc
   # is called the first time by initBaseLexer()
-  assert(L.sentinel < L.bufLen)
-  toCopy = L.bufLen - L.sentinel - 1
+  assert(L.sentinel + 1 <= L.buf.len)
+  toCopy = L.buf.len - (L.sentinel + 1)
   assert(toCopy >= 0)
   if toCopy > 0:
     when defined(js):
```

Comment:
`when defined(js) or nimvm:` to avoid code duplication

Reply:
`nimvm` unfortunately cannot be used that way: https://nim-lang.org/docs/manual.html#statements-and-expressions-when-nimvm-statement

Reply:
@Vindaar Would it be possible to nest the when statements?